f5/bigip/__init__.py

```diff
@@ -65,6 +66,25 @@ def hostname(self):
     def icontrol_version(self):
         return self._meta_data['icontrol_version']
 
+    @property
+    def tmos_version(self):
+        # We return tmos_version, or if None,
+        # we grab it and return value via _get_tmos_version(),
+        # at the same time updating _meta_data['tmos_version']
+        return self._meta_data['tmos_version'] or self._get_tmos_version()
+
+    def _get_tmos_version(self):
+        connect = self._meta_data['bigip']._meta_data['icr_session']
+        # Need to add this due to BIGIP class URI manipulation
+        if self._meta_data['uri'].endswith('tm/'):
+            base_uri = self._meta_data['uri'] + 'sys/'
+        else:
+            base_uri = self._meta_data['uri'] + 'tm/sys/'
```

Comment:
This is not for tests onlt. This will break backwards compatibility for EVERY use of BigIP..meaning all solutions that use old syntax will get Uri errors..as get tmos version method will fail as it adds tm/sys into Metadata URI